test/e2e/common.go

```diff
@@ -47,22 +47,27 @@ const (
 var m sync.Mutex
 
 type e2eTest struct {
-	env env
-	kn  kn
+	env                    env
+	kn                     kn
+	createNamespaceOnSetup bool
+	namespaceCreated       bool
 }
 
 func NewE2eTest(t *testing.T) *e2eTest {
 	return &e2eTest{
-		env: buildEnv(t),
+		env:                    buildEnv(t),
+		createNamespaceOnSetup: true,
 	}
 }
 
 // Setup set up an environment for kn integration test returns the Teardown cleanup function
 func (test *e2eTest) Setup(t *testing.T) {
 	test.env.Namespace = fmt.Sprintf("%s%d", test.env.Namespace, getNamespaceCountAndIncrement())
 	test.kn = kn{t, test.env.Namespace, Logger{}}
-	test.CreateTestNamespace(t, test.env.Namespace)
-	time.Sleep(20 * time.Second)
+	if test.createNamespaceOnSetup {
+		test.CreateTestNamespace(t, test.env.Namespace)
+		time.Sleep(20 * time.Second)
```

Comment:
why `time.Sleep`?

Reply:
We always had that sleep since namespaces take time to create. See previous code.

We can experiment with smaller values but that's a separate PR.

I had to make namespace creation optional since this plugin e2e would have created a namespace but never used it.

Reply:
Was in previous code so kept it. I think was there to give time for namespace to be created.

I think we should probably reduce the time since NS creation might be faster. However, that's a separate PR.

I'll open an issue to try to get e2e tests faster and point to that line as one place.

Reply:
I created issue #598